Bottom: 2bd50c180d0a8c3d6f476336bb92049576b654cc
Top:    e25907e4f594936c2503179c812213576c963fab
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-29 17:07:49 +0100

Refresh of hook-up-debug-drawing-apis-for

---

diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index b5e28b6..3d81dc7 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -1230,13 +1230,19 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
 HID *
 ghid_request_debug_draw (void)
 {
+  /* No special setup requirements, drawing goes into
+   * the backing pixmap. */
+  return gport;
 }
 
 void
 ghid_flush_debug_draw (void)
 {
+  ghid_screen_update ();
 }
 
 ghid_finish_debug_draw (void)
 {
+  ghid_flush_debug_draw ();
+  /* No special tear down requirements */
 }
